vyperlang/vyper
# chore: add sharding for CI tests

### What I did
use pytest-split to split fuzzer tests across multiple runners

### How I did it
followed https://github.com/yearn/yearn-vaults/blob/cdee41137d37af933db686fae51641f5f4521956/.github/workflows/test.yaml#L58-L96

### How to verify it
check multiple fuzzer runners take less time than single runner. see https://github.com/charles-cooper/vyper/actions/runs/2031458674

note on timing runs:
20 runners: ~5 minute avg runtime (https://github.com/charles-cooper/vyper/actions/runs/2030982431)
16 runners: ~6 minute avg runtime (https://github.com/charles-cooper/vyper/actions/runs/2031458674)
10 runners: ~8 minute avg runtime (https://github.com/charles-cooper/vyper/actions/runs/2029249072)
8 runners: ~9 minute avg runtime (https://github.com/charles-cooper/vyper/actions/runs/2031018360)
1 runner: ~1hr runtime (https://github.com/charles-cooper/vyper/actions/runs/2026600464)

without doing actual linear regression, i think the overhead for each runner is about ~2 minutes. so we see diminishing returns around 10 runners. but as we add more fuzzer tests (which we will in https://github.com/vyperlang/vyper/pull/2715), it makes sense to add more runners

### Description for the changelog

### Cute Animal Picture

![Put a link to a cute animal picture inside the parenthesis-->]()
